src/altinn-app-frontend/src/components/presentation/NavBar.test.tsx

```diff
@@ -1,26 +1,70 @@
+import { render, screen } from '@testing-library/react';
+import { getLanguageFromCode } from 'altinn-shared/language';
+import userEvent from '@testing-library/user-event';
+import { IAppLanguage } from 'altinn-shared/types';
+
+const renderNavBar = (props?: Partial<INavBarProps>) => {
+  const mockClose = jest.fn();
+  const mockBack = jest.fn();
+  const mockAppLanguageChange = jest.fn();
+  const mockAppLanguages: IAppLanguage[] = [
+    {
+      language: 'nb',
+    },
+    {
+      language: 'en',
+    },
+  ];
+  render(
+    <NavBar
+      language={getLanguageFromCode('nb')}
+      handleClose={mockClose}
+      handleBack={mockBack}
+      appLanguages={mockAppLanguages}
+      onAppLanguageChange={mockAppLanguageChange}
+      selectedAppLanguage={'nb'}
+      {...props}
+    />,
+  );
+
+  return { mockClose, mockBack, mockAppLanguageChange };
+};
 
 describe('components/presentation/NavBar.tsx', () => {
-  it('should render back button by default', () => {
-    const wrapper = mount(
-      <NavBar language={{}} handleClose={null} handleBack={null} />,
-    );
-    expect(wrapper.find('div.a-modal-navbar')).toHaveLength(1);
-    expect(wrapper.find('button.a-modal-close')).toHaveLength(1);
+  it('should render close button', async () => {
+    const { mockClose } = renderNavBar();
+    const closeButton = screen.getByRole('button', { name: /Lukk Skjema/i });
+    await userEvent.click(closeButton);
+    expect(mockClose).toHaveBeenCalled();
+  });
+
+  it('should hide close button', () => {
+    renderNavBar({ hideCloseButton: true });
+    expect(screen.queryByRole('button', { name: /Lukk Skjema/i })).toBeNull();
   });
 
-  it('should not render back button when hideBackButton is supplied', () => {
-    const wrapper = mount(
-      <NavBar
-        language={{}}
-        handleClose={null}
-        handleBack={null}
-        hideCloseButton={true}
-      />,
-    );
-    expect(wrapper.find('div.a-modal-navbar')).toHaveLength(1);
-    expect(wrapper.find('button.a-modal-close')).toHaveLength(0);
+  it('should render back button', async () => {
+    const { mockBack } = renderNavBar({ showBackArrow: true });
+    const backButton = screen.getByRole('button', { name: /Tilbake/i });
+    await userEvent.click(backButton);
+    expect(mockBack).toHaveBeenCalled();
+  });
+  it('should not render back button', async () => {
+    renderNavBar();
+    expect(screen.queryByRole('button', { name: /Tilbake/i })).toBeNull();
+  });
+  it('should render app language', async () => {
+    const { mockAppLanguageChange } = renderNavBar({
+      showLanguageSelector: true,
+    });
+    const dropdown = screen.getByRole('combobox', { name: /Språk/i });
+    await userEvent.selectOptions(dropdown, 'English');
+    expect(mockAppLanguageChange).toHaveBeenCalledWith('en');
+  });
+  it('should not render app language combobox', async () => {
+    renderNavBar();
+    expect(screen.queryByRole('combobox', { name: /Språk/i })).toBeNull();
```

Comment:
This test is also a bit too explicit.
We might also go away from having this text localised and only use the English ('Language') version.
If I do not understand the language on the page in the first place, and in need to change the language, I would probably not understand the word for language in that language either.